File: anul 2/semestrul 1/poo/kotlin/src/lab5/lab5_2.kt
package lab5

open class Employee(var name: String, var baseSalary: Double){
    open fun printDetails(){
        println("Nume: $name,Rol: Angajat, Salariu de baza: $baseSalary")
    }
    open fun salaryCalculator(){
        println("Calcul salariu pentru Angajat: $baseSalary")
    }

}
open class Manager(name: String, baseSalary: Double, private var mangerBonus:Double): Employee(name, baseSalary) {
    override fun printDetails() {
        println("Nume: $name,Rol: Manager, Salariu de baza: $baseSalary")
    }
    override fun salaryCalculator() {
        println("Calcul salariu pentru Manager: ${baseSalary*mangerBonus}")
    }
}
class Director(name: String,baseSalary: Double,private var directorBonus:Double,private var department:String):Manager(name,baseSalary,directorBonus) {
    override fun printDetails() {
        println("Nume: $name,Rol: Director, Departament: $department, Salariu de baza: $baseSalary")
    }

    override fun salaryCalculator() {
        println("Calcul salariu pentru Director: ${baseSalary * directorBonus}")
    }

}
fun main() {
    val employee = Employee("Florin",2600.0)
    val manager = Manager("Andrei",2600.0,1.5)
    val director = Director("Daniel",2600.0,3.5,"Hr")

    employee.printDetails()
    employee.salaryCalculator()
    manager.printDetails()
    manager.salaryCalculator()
    director.printDetails()
    director.salaryCalculator()
}

File: anul 2/semestrul 1/poo/lab5/lab5_2.cpp
#include <iostream>
#include <string>

using namespace std;


class Angajat {
protected:
    string nume;
    double salariuBaza;

public:
    Angajat(const string& nume, double salariuBaza)
        : nume(nume), salariuBaza(salariuBaza) {}

    virtual void afiseazaDetalii() const {
        cout << "Nume: " << nume << ", Salariu de baza: " << salariuBaza << endl;
    }

    virtual void calculeazaSalariu() {
        cout << "Calcul salariu pentru Angajat: " << salariuBaza << endl;
    }
};


class Manager : public Angajat {
protected:
    double bonusManager;

public:
    Manager(const string& nume, double salariuBaza, double bonusManager)
        : Angajat(nume, salariuBaza), bonusManager(bonusManager) {}

    void afiseazaDetalii() const override {
        cout << "Nume: " << nume << ", Salariu de baza: " << salariuBaza << ", Bonus Manager: " << bonusManager << endl;
    }

    void calculeazaSalariu() override {
        double salariuTotal = salariuBaza + bonusManager;
        cout << "Calcul salariu pentru Manager: " << salariuTotal << endl;
    }
};


class Director : public Manager {
protected:
    double bonusDirector;

public:
    Director(const string& nume, double salariuBaza, double bonusManager, double bonusDirector)
        : Manager(nume, salariuBaza, bonusManager), bonusDirector(bonusDirector) {}

    void afiseazaDetalii() const override {
        cout << "Nume: " << nume << ", Salariu de baza: " << salariuBaza << ", Bonus Manager: " << bonusManager << ", Bonus Director: " << bonusDirector << endl;
    }

    void calculeazaSalariu() override {
        double salariuTotal = salariuBaza + bonusManager + bonusDirector;
        cout << "Calcul salariu pentru Director: " << salariuTotal << endl;
    }
};

int main() {
    Angajat angajat("Ion", 50000);
    Manager manager("Andrei", 60000, 10000);
    Director director("David", 70000, 10000, 20000);


    angajat.afiseazaDetalii();
    angajat.calculeazaSalariu();

    manager.afiseazaDetalii();
    manager.calculeazaSalariu();

    director.afiseazaDetalii();
    director.calculeazaSalariu();

    return 0;
}
